app/models/result.rb

```diff
@@ -129,7 +129,15 @@ def to_s
     end
   end
 
-  def self.create_new!(competitor, new_place, result_type, result_subtype = nil)
+  def self.create_new_results!(placed_competitors, result_type, result_subtype = nil)
+    results = placed_competitors.compact.map do |placed_competitor|
+      create_new_hashed_result(placed_competitor[:competitor], placed_competitor[:new_place], result_type, result_subtype)
+    end
+
+    Result.insert_all(results.compact)
+  end
+
+  private_class_method def self.create_new_hashed_result(competitor, new_place, result_type, result_subtype = nil)
```

Comment:
I don't like introducing a new pattern to the codebase. I do see one other place where we needed this, (see `minute_second_formatter.rb`, and do that?